File: scala/src/main/scala/TimeUtility.scala
import java.time.LocalTime

object TimeUtility {
  def timeOfDay: String = {
    LocalTime.now() match {
      case night if night.getHour >= 0 && night.getHour < 6 => "Night"
      case morning if morning.getHour >= 6 && morning.getHour < 12 => "Morning"
      case afternoon if afternoon.getHour >= 12 && afternoon.getHour < 18 => "Afternoon"
      case _ => "Evening"
    }
  }
}

File: java/src/main/java/unit/tests/intro/TimeUtility.java
package unit.tests.intro;

import static java.time.LocalTime.now;

public class TimeUtility {
    public String getTimeOfDay() {
        var time = now();

        if (time.getHour() < 6) {
            return "Night";
        } else if (time.getHour() < 12) {
            return "Morning";
        } else if (time.getHour() < 18) {
            return "Afternoon";
        }
        return "Evening";
    }
}